cooperative_lanechange/src/cooperative_lanechange_node.cpp

```diff
@@ -0,0 +1,664 @@
+/*
+ * Copyright (C) 2019-2022 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include "cooperative_lanechange/cooperative_lanechange_node.hpp"
+
+namespace cooperative_lanechange
+{
+  namespace std_ph = std::placeholders;
+
+  CooperativeLaneChangePlugin::CooperativeLaneChangePlugin(const rclcpp::NodeOptions &options)
+      : carma_guidance_plugins::TacticalPlugin(options)
+  {
+    // Create initial config
+    config_ = Config();
+
+    // Declare parameters
+    config_.trajectory_time_length = declare_parameter<double>("trajectory_time_length", config_.trajectory_time_length);
+    config_.control_plugin_name = declare_parameter<std::string>("control_plugin_name", config_.control_plugin_name);
+    config_.minimum_speed = declare_parameter<double>("minimum_speed", config_.minimum_speed);
+    config_.max_accel = declare_parameter<double>("max_accel", config_.max_accel);
+    config_.minimum_lookahead_distance = declare_parameter<double>("minimum_lookahead_distance", config_.minimum_lookahead_distance);
+    config_.maximum_lookahead_distance = declare_parameter<double>("maximum_lookahead_distance", config_.maximum_lookahead_distance);
+    config_.minimum_lookahead_speed = declare_parameter<double>("minimum_lookahead_speed", config_.minimum_lookahead_speed);
+    config_.maximum_lookahead_speed = declare_parameter<double>("maximum_lookahead_speed", config_.maximum_lookahead_distance);
```

Comment:
config_.maximum_lookahead_distance should be speed

Reply:
Good catch! Fixed.